workers/unity/Packages/com.improbable.gdk.buildsystem/Configuration/SpatialOSBuildConfiguration.cs

```diff
@@ -6,138 +6,93 @@
 
 namespace Improbable.Gdk.BuildSystem.Configuration
 {
-    [CreateAssetMenu(fileName = "SpatialOS Build Configuration", menuName = CreateMenuPath)]
-    public class SpatialOSBuildConfiguration : SingletonScriptableObject<SpatialOSBuildConfiguration>
-    {
-        internal const string CreateMenuPath = "SpatialOS/Build Configuration";
-
+    [CreateAssetMenu(fileName = "SpatialOS Build Configuration", menuName = EditorConfig.BuildConfigurationMenu)]
+    public class SpatialOSBuildConfiguration : ScriptableSingleton<SpatialOSBuildConfiguration>
+    {        
+        [SerializeField] public List<WorkerBuildConfiguration> WorkerBuildConfigurations = 
+            new List<WorkerBuildConfiguration>();
+        
         [SerializeField] private bool isInitialised;
+        
+        public BuildEnvironmentConfig GetEnvironmentConfigForWorker(string workerType, BuildEnvironment environment)
+        {
+            var config = WorkerBuildConfigurations.FirstOrDefault(x => x.WorkerType == workerType);
+            if (config == null)
+            {
+                throw new ArgumentException($"Unknown worker type {workerType}.");
+            }
 
-        [SerializeField] public List<WorkerBuildConfiguration> WorkerBuildConfigurations;
+            return config.GetEnvironmentConfig(environment);
+        }
 
-        public override void OnEnable()
+        internal string[] GetScenePathsForWorker(string workerType)
+        {
+            return GetScenesForWorker(workerType)
+                .Where(sceneAsset => sceneAsset != null)
+                .Select(AssetDatabase.GetAssetPath)
+                .ToArray();
+        }
+        
+        internal void UpdateEditorScenesForBuild()
         {
-            base.OnEnable();
+            EditorApplication.delayCall += () =>
```

Comment:
This delay should be wherever this function is called, not in the function itself

Reply:
this was not changed and has been that way from the very beginning (it only looks that way because of github 🤷‍♀️ ). should we remove the function completely?

Reply:
For a bit of context for this method, it was necessary to update scenes for build because we'd do scene loading using the scene name / scene index e.g. for pirates tutorial, and "play in editor" would break. Builds are not affected by this because they can be specified as part of build arguments.

If we remove it we may need to add a note in documentation to say users will need to ensure their scenes will need to be included in build settings if they want to do some scene loading tricks for "play in editor".

Right now we don't do scene loading per worker (we do "scene prefab instantiation" which doesn't need this)